google-youtube-upload.html

```diff
@@ -21,447 +30,687 @@
-        this._uploadStartTime = Date.now();
-        uploader.upload();
-      },
-
-      _handleFileChanged: function() {
-        this._selectedFile = this.$.file.files[0];
+          this._tearDownMediaRecorder();
+        }.bind(this));
+      }.bind(this));
+    },
 
-        if (this.auto) {
-          this.uploadFile(this._selectedFile);
-        }
-      },
+    _tearDownMediaRecorder: function() {
+      this._recordingState = 'stopped';
 
-      _handleUploadClicked: function() {
-        this.uploadFile(this._selectedFile);
-      },
+      if (this._mediaRecorder) {
+        this._mediaRecorder.stream.getAudioTracks().forEach(function(track) {
+          track.stop();
+        });
+        this._mediaRecorder.stream.getVideoTracks().forEach(function(track) {
+          track.stop();
+        });
+        this._mediaRecorder = null;
+        this._recordedBlobs = null;
+      }
+    },
 
-      _pollForVideoStatus: function() {
+    _loadChannels: function() {
+      if (gapi && gapi.client && this._accessToken && !this._loadChannelRequested) {
+        this._loadChannelRequested = true;
         gapi.client.request({
-          path: '/youtube/v3/videos',
+          path: '/youtube/v3/channels',
           params: {
-            part: 'status',
-            id: this.videoId
+            part: 'snippet',
+            mine: true
           },
           callback: function(response) {
             if (response.error) {
-              // Not exactly sure how to handle this one, since it means the status polling failed.
-              setTimeout(this._pollForVideoStatus.bind(this), STATUS_POLLING_ITERVAL_MILLIS);
+              this.fire('youtube-upload-fail', response.error.message);
+              this._channel = null;
             } else {
-              var status = response.items[0].status;
-
-              switch (status.uploadStatus) {
-                // This is a non-final status, so we need to poll again.
-                case 'uploaded':
-                  this.fire('youtube-processing-poll', status);
-                  setTimeout(this._pollForVideoStatus.bind(this), STATUS_POLLING_ITERVAL_MILLIS);
-                break;
-
-                // The video was successfully transcoded and is available.
-                case 'processed':
-                  this.fire('youtube-processing-complete', this.videoId);
-                break;
-
-                // All other statuses indicate a permanent transcoding failure.
-                default:
-                  this.fire('youtube-processing-fail', status);
-                break;
-              }
+              this._channel = {
+                name: response.items[0].snippet.title,
+                thumbnail: response.items[0].snippet.thumbnails.default.url
+              };
             }
           }.bind(this)
         });
       }
-    });
-  })();
+    },
+
+    _pollForVideoStatus: function() {
```

Comment:
too bad the API doesn't have a better approach than polling. I remember the same type of thing with the drive zip download endpoint :\